programs/locked-voter/src/instructions/approve_move_request.rs

```diff
@@ -0,0 +1,65 @@
+use crate::*;
+/// Accounts for [voter::approve_move_request].
+#[derive(Accounts)]
+pub struct ApproveMoveRequest<'info> {
+    /// [Request].
+    #[account(
+        has_one = escrow,
+        has_one = recovery_key,
+    )]
+    pub request: Box<Account<'info, MoveRequest>>,
+    /// [Old_Escrow].
+    #[account(mut, has_one = locker)]
+    pub escrow: Box<Account<'info, Escrow>>,
+
+    /// CHECK: checked by request account
+    pub recovery_key: UncheckedAccount<'info>,
+
+    // (old) => Locker => Governor => Smart Wallet (Sends this IX)
+    #[account(has_one = governor)]
+    pub locker: Box<Account<'info, Locker>>,
+
+    #[account(has_one = smart_wallet)]
+    pub governor: Box<Account<'info, Governor>>,
+
+    pub smart_wallet: Signer<'info>,
+}
+
+impl<'info> ApproveMoveRequest<'info> {
+    pub fn approve_move_request(&mut self) -> Result<()> {
+        self.escrow.recovery_key = self.recovery_key.key();
+        self.escrow.freeze_timestamp = Clock::get()?.unix_timestamp;
```

Comment:
it is weird that freeze_timestamp is being used to determine whether the escrow is locked or not,
based on the function: escrow.is_frozen(), even after the multisig approve this request, after a duration, user can still interact with this escrow